dashboard/app/helpers/levels_helper.rb

```diff
@@ -45,8 +45,8 @@ def build_script_level_url(script_level, params = {})
     url_from_path(build_script_level_path(script_level, params))
   end
 
-  def url_from_path(path)
-    CDO.studio_url(path)
+  def url_from_path(path, scheme = '')
+    CDO.studio_url(path, scheme)
```

Comment:
great find! sorry to drag this out but it seems like using `CDO.default_scheme` here might solve the local development problem!

* https://github.com/code-dot-org/code-dot-org/blob/a5123e2575388295107a17757c3de8171dd51ab0/config.yml.erb#L331
* https://github.com/code-dot-org/code-dot-org/blob/365570bd1cc7ddc62e84762dce7d18dd2a610a20/dashboard/app/controllers/api_controller.rb#L593

Sorry I missed that earlier.

Reply:
Oh that's much better, thanks for finding that